src/allmydata/test/cli/test_alias.py

```diff
@@ -1,105 +1,126 @@
 import json
-from mock import patch
 
 from twisted.trial import unittest
 from twisted.internet.defer import inlineCallbacks
 
-from allmydata.util.encodingutil import unicode_to_argv
 from allmydata.scripts.common import get_aliases
 from allmydata.test.no_network import GridTestMixin
 from .common import CLITestMixin
-from ..common_util import skip_if_cannot_represent_argv
+from allmydata.util import encodingutil
 
 # see also test_create_alias
 
 class ListAlias(GridTestMixin, CLITestMixin, unittest.TestCase):
 
     @inlineCallbacks
-    def test_list(self):
-        self.basedir = "cli/ListAlias/test_list"
+    def _test_list(self, alias, encoding):
```

Comment:
Nice helper with nice documentation :)

Just a minor comment on style, feel free to ignore it as I am just preaching here :)

maybe rename to `checkCreateAlias()` 

As a convention, when a method inside the test code is private, it means that the method should not be used in another test... and it might depend on some setup/teardown done by the test callling this private test helper.

For specialized assert helper use public names and this means that anyone can (re)use that assertion.

And if the helper method takes an expected value called it `assertSomething` otherwise if it only has input call it `checkSomething`.

Reply:
I like this taxonomy, thanks for suggesting it.